src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/RestoreOperation/RestoreDatabaseHelper.cs

```diff
@@ -32,20 +32,38 @@ public class RestoreDatabaseHelper
         public RestoreConfigInfoResponse CreateConfigInfoResponse(RestoreConfigInfoRequestParams restoreConfigInfoRequest)
         {
             RestoreConfigInfoResponse response = new RestoreConfigInfoResponse();
-            RestoreDatabaseTaskDataObject restoreTaskObject = CreateRestoreForNewSession(restoreConfigInfoRequest.OwnerUri);
-            if (restoreTaskObject != null)
+            RestoreDatabaseTaskDataObject restoreTaskObject = null;
+            
+            try
             {
-                // Default Data folder path in the target server
-                response.ConfigInfo.Add(RestoreOptionsHelper.DataFileFolder, restoreTaskObject.DefaultDataFileFolder);
-                // Default log folder path in the target server
-                response.ConfigInfo.Add(RestoreOptionsHelper.LogFileFolder, restoreTaskObject.DefaultLogFileFolder);
-                // The db names with backup set
-                response.ConfigInfo.Add(RestoreOptionsHelper.SourceDatabaseNamesWithBackupSets, restoreTaskObject.GetDatabaseNamesWithBackupSets());
-                // Default backup folder path in the target server
-                response.ConfigInfo.Add(RestoreOptionsHelper.DefaultBackupFolder, restoreTaskObject.DefaultBackupFolder);
-            }
+                restoreTaskObject = CreateRestoreForNewSession(restoreConfigInfoRequest.OwnerUri);
+                if (restoreTaskObject != null)
+                {
+                    // Default Data folder path in the target server
+                    response.ConfigInfo.Add(RestoreOptionsHelper.DataFileFolder, restoreTaskObject.DefaultDataFileFolder);
+                    // Default log folder path in the target server
+                    response.ConfigInfo.Add(RestoreOptionsHelper.LogFileFolder, restoreTaskObject.DefaultLogFileFolder);
+                    // The db names with backup set
+                    response.ConfigInfo.Add(RestoreOptionsHelper.SourceDatabaseNamesWithBackupSets, restoreTaskObject.GetDatabaseNamesWithBackupSets());
+                    // Default backup folder path in the target server
+                    response.ConfigInfo.Add(RestoreOptionsHelper.DefaultBackupFolder, restoreTaskObject.DefaultBackupFolder);
+                }
 
+            }
+            catch(Exception ex) 
+            {
+                Logger.Write(LogLevel.Warning, $"Failed to create restore config info. error: { ex.Message}");
+            }
+            finally
+            {
+                if(restoreTaskObject != null && restoreTaskObject.Server != null && restoreTaskObject.Server.ConnectionContext != null && 
```

Comment:
suggestion: you might be able to use `var context = restoreTaskObject?.Server?.ConnectionContext` and then check `if (context?.IsOpen)` to simplify this a bit as it's a ton of null checks. Plus then you can just do `context.Disconnect()` in the main block which is nice.